code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InternalMessagingUtils.java

```diff
@@ -647,6 +661,16 @@ private static boolean isPushTokenSyncTimeoutExpired(final long eventTimestamp)
     // ========================================================================================
     // Datastore utils
     // ========================================================================================
+    /**
+     * Persists the provided push token in the Messaging extension {@link NamedCollection}.
+     *
+     * @param pushToken A {@code String} containing the push token to be persisted
+     */
+    static void persistPushToken(final String pushToken) {
+        final NamedCollection messagingNamedCollection = getNamedCollection();
+        messagingNamedCollection.setString(
+                MessagingConstants.NamedCollectionKeys.Messaging.PUSH_IDENTIFIER, pushToken);
```

Comment:
nit: It is safer to check if `pushToken` is null and if so call `NamedCollection.remove()` instead. It is true that the underlying `SharedPreferencesEditor.putString` will remove the value if passed null, but `NamedCollection.setString` doesn't explicitly state that behavior. My concern is if the `NamedCollection` implementation is changed then passing null may no longer remove the value.